scripts/sanitycheck

```diff
@@ -3486,60 +3489,142 @@ def create_gcda_files(extracted_coverage_info):
         with open(filename, 'wb') as fp:
             fp.write(bytes.fromhex(hexdump_val))
 
-def generate_coverage(outdir, ignores):
 
-    for filename in glob.glob("%s/**/handler.log" %outdir, recursive=True):
-        gcov_data = retrieve_gcov_data(filename)
-        capture_complete = gcov_data['complete']
-        extracted_coverage_info = gcov_data['data']
-        if capture_complete:
-            create_gcda_files(extracted_coverage_info)
-            verbose("Gcov data captured: {}".format(filename))
-        else:
-            error("Gcov data capture incomplete: {}".format(filename))
+class CoverageTool(object):
+    """ Base class for every supported coverage tool
+    """
+
+    def __init__(self):
+        self.gcov_tool = options.gcov_tool
+
+    @staticmethod
+    def factory(tool):
+        if tool == 'lcov':
+            return Lcov()
+        if tool == 'gcovr':
+            return Gcovr()
+        error("Unsupported coverage tool specified: {}".format(tool))
+
+    def generate(self, outdir):
+        for filename in glob.glob("%s/**/handler.log" % outdir, recursive=True):
+            gcov_data = retrieve_gcov_data(filename)
```

Comment:
can we move this function into the class as well, maybe as a static method, this way we will have all things related to coverage under one class.

Reply:
done